application/Modules/Clipboard/Producers/ClipboardCopyEventProducer.cs

```diff
@@ -0,0 +1,240 @@
+                windowClass.hbrBackground = IntPtr.Zero;
+                windowClass.lpszMenuName = "";
+                windowClass.lpszClassName = className;
+
+
+                ClipboardNativeMethods.RegisterClass(ref windowClass);
+
+                // Creates window to register clipboard update messages
+                WindowHandle = ClipboardNativeMethods.CreateWindowEx(0,
+                                                                     className,
+                                                                     "",
+                                                                     0,
+                                                                     0, 0,
+                                                                     1, 1,
+                                                                     IntPtr.Zero,
+                                                                     IntPtr.Zero,
+                                                                     IntPtr.Zero,
+                                                                     IntPtr.Zero);
+
+                ClipboardNativeMethods.AddClipboardFormatListener(WindowHandle);
+            }
+
+            /// <summary>
+            ///     The underlying window handle
+            /// </summary>
+            private IntPtr WindowHandle { get; }
+
+            /// <summary>
+            ///     Event invoked when clipboard is updated
+            /// </summary>
+            public event ClipboardEventHandler? ClipboardUpdated;
+
+            private IntPtr OnClipboardUpdate(IntPtr hWnd, uint messageId, IntPtr wParam, IntPtr lParam)
+            {
+                if (messageId == (int) NativeMethods.MessageType.WM_CLIPBOARDUPDATE)
+                {
+                    ClipboardUpdated?.Invoke(hWnd, messageId, wParam, lParam);
+                }
+
+                return ClipboardNativeMethods.DefWindowProc(hWnd, messageId, wParam, lParam);
+            }
+
+            #region Dispose
+
+            private bool isDisposed;
+
+            /// <summary>
+            ///     Frees all unmanaged resources
+            /// </summary>
+            public void Dispose()
+            {
+                Dispose(true);
+            }
+
+            ~ClipboardWindowMessageSink()
+            {
+                Dispose(false);
+            }
+
+            private void Dispose(bool disposing)
+            {
+                if (isDisposed || !disposing)
+                {
+                    return;
+                }
+
+                isDisposed = true;
+                ClipboardNativeMethods.RemoveClipboardFormatListener(WindowHandle);
+                ClipboardNativeMethods.DestroyWindow(WindowHandle);
+                internalWindowMessageHandler = null;
+            }
+
+            #endregion
+        }
+
+        #endregion
+
+        #region Clipboard native methods
+
+        internal static class ClipboardNativeMethods
```

Comment:
The content of this is not specific to the clipboard module. I would suggest moving it into the shared `NativeMethods` class.